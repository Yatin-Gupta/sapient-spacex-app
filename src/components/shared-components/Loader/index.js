import React from "react";

/**
 * Loader Component with white overlay
 * @param {*} props
 */
function Loader(props) {
  return props.show ? (
    <div className="overlay">
      <div className="loader" data-testid="loader"></div>
    </div>
  ) : (
    <React.Fragment />
  );
}

export default Loader;
