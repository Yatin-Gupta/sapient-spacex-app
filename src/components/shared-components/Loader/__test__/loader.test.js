import React from "react";
import ReactDOM from "react-dom";
import Loader from "../index";

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("Loader component renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Loader />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("Loader component should renders when show props is set", () => {
  const { getByTestId } = render(<Loader show={true} />);
  expect(getByTestId("loader")).toHaveClass("loader");
});

it("Loader component should not renders when show props is unset", () => {
  const { queryByTestId } = render(<Loader show={false} />);
  expect(queryByTestId("loader")).toBeNull();
});

it("Loader component matches the snapshot", () => {
  const tree = renderer.create(<Loader show={true} />).toJSON();
  expect(tree).toMatchSnapshot();
});
