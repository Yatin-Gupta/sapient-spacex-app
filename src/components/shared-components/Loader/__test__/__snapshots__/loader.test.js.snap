// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component matches the snapshot 1`] = `
<div
  className="overlay"
>
  <div
    className="loader"
    data-testid="loader"
  />
</div>
`;
